crates/ruff_linter/src/rules/flake8_pyi/rules/unused_private_type_definition.rs

```diff
@@ -210,13 +223,18 @@ pub(crate) fn unused_private_type_var(
             continue;
         };
 
-        diagnostics.push(Diagnostic::new(
+        let mut diagnostic = Diagnostic::new(
             UnusedPrivateTypeVar {
                 type_var_like_name: id.to_string(),
                 type_var_like_kind: type_var_like_kind.to_string(),
             },
             binding.range(),
-        ));
+        );
+
+        let edit = fix::edits::delete_stmt(stmt, None, checker.locator(), checker.indexer());
+        diagnostic.set_fix(Fix::safe_edit(edit));
+
+        diagnostics.push(diagnostic);
```

Comment:
a couple of small things:
1. Let's first make this fix [preview](https://docs.astral.sh/ruff/preview/)-only. If it's been a few months and we haven't had any bug reports about it, we can promote it so that it's also available in stable mode
2. I think this fix has to be marked as unsafe, unfortunately. There's always a chance that somebody is importing the TypeVar from another module, even if it's marked as private, and this fix would break your code if you're doing that.

```suggestion
        if checker.settings.preview.is_enabled() {
	        let edit = fix::edits::delete_stmt(stmt, None, checker.locator(), checker.indexer());
	        diagnostic.set_fix(Fix::unsafe_edit(edit));
	    }

        diagnostics.push(diagnostic);
```

This might mean that you have to change the `FIX_AVAILABILITY` constant on line 34 to `FIX_AVAILABILITY::SOMETIMES`